snapshotter/snapshotter.py

```diff
@@ -378,6 +394,13 @@ def snapshot(source, dest, debug=False, min_snapshots=3, extra_args=None):
     """
     date = _datetime()
     user, host, snapshots_root = _parse_path(dest)
+    if max_snapshots > -1 and max_snapshots - min_snapshots < 0:
+        raise InconsistentArgumentsError(
+            "--max-snapshots must be greater than --min-snapshots")
+    if max_snapshots > -1 and max_snapshots + 1 - min_snapshots >= 0:
+        snapshots = _ls_snapshots(dest)
+        for i in range(len(snapshots) - max_snapshots + 1):
+            _rm(snapshots[i], user, host, directory=True, debug=debug)
```

Comment:
Hmm, I like the idea of using infinity as the default value. It removes any need for a "special case" where we try to avoid enforcing max snapshots if the user didn't specify it, we just set it to infinity by default and then go ahead and enforce it blindly knowing that by default it won't do anything. Nice.

I think `float("inf")` might be better than `sys.maxint` though? It's not an int, but I don't think that matters, it's a number that's always greater than any other number in `>` and `<` comparisons.

```python
In [7]: float('inf') > 1000000
Out[7]: True

In [8]: float('inf') > sys.maxint
Out[8]: True
```